Library/String+SimpleHTML.swift

```diff
@@ -21,44 +21,55 @@ public extension String {
+      let italic = optionalItalic ?? [NSAttributedString.Key.font: baseFont.italicized]
+
+      guard let data = self.data(using: String.Encoding.utf8) else { return nil }
+
+      let options: [NSAttributedString.DocumentReadingOptionKey: Any] = [
+        NSAttributedString.DocumentReadingOptionKey.documentType: NSAttributedString.DocumentType.html,
+        NSAttributedString.DocumentReadingOptionKey.characterEncoding: String.Encoding.utf8.rawValue
+      ]
+      guard let string = try? NSMutableAttributedString(data: data, options: options, documentAttributes: nil)
+      else {
+        return nil
+      }
 
-    // If bold or italic are not specified we can derive them from `font`.
-    let bold = optionalBold ?? [NSAttributedString.Key.font: baseFont.bolded]
-    let italic = optionalItalic ?? [NSAttributedString.Key.font: baseFont.italicized]
+      // Sub all bold and italic fonts in the attributed html string
+      let stringRange = NSRange(location: 0, length: string.length)
+      string.beginEditing()
+      string
+        .enumerateAttribute(NSAttributedString.Key.font, in: stringRange, options: []) { value, range, _ in
 
-    guard let data = self.data(using: String.Encoding.utf8) else { return nil }
+          guard let htmlFont = value as? UIFont else { return }
+          let newAttributes: Attributes
 
-    let options: [NSAttributedString.DocumentReadingOptionKey: Any] = [
-      NSAttributedString.DocumentReadingOptionKey.documentType: NSAttributedString.DocumentType.html,
-      NSAttributedString.DocumentReadingOptionKey.characterEncoding: String.Encoding.utf8.rawValue
-    ]
-    guard let string = try? NSMutableAttributedString(data: data, options: options, documentAttributes: nil)
-    else {
-      return nil
-    }
+          if htmlFont.fontDescriptor.symbolicTraits.contains(.traitBold) {
+            newAttributes = bold
+          } else if htmlFont.fontDescriptor.symbolicTraits.contains(.traitItalic) {
+            newAttributes = italic
+          } else {
+            newAttributes = base
+          }
 
-    // Sub all bold and italic fonts in the attributed html string
-    let stringRange = NSRange(location: 0, length: string.length)
-    string.beginEditing()
-    string.enumerateAttribute(NSAttributedString.Key.font, in: stringRange, options: []) { value, range, _ in
+          string.addAttributes(newAttributes, range: range)
+        }
+      string.endEditing()
 
-      guard let htmlFont = value as? UIFont else { return }
-      let newAttributes: Attributes
+      return string
+    }
 
-      if htmlFont.fontDescriptor.symbolicTraits.contains(.traitBold) {
-        newAttributes = bold
-      } else if htmlFont.fontDescriptor.symbolicTraits.contains(.traitItalic) {
-        newAttributes = italic
-      } else {
-        newAttributes = base
+    if Thread.isMainThread {
+      return parsedHtml()
+    } else {
+      return DispatchQueue.main.sync {
```

Comment:
Shouldn't this be `async`? 🤔

Reply:
We actually do want this to run synchronously on the main thread. The difference between `async` and `sync` here is essentially that the code after the closure won't be executed until the closure returns, this is also why we're able to just do return `return DispatchQueue.main.sync { ... }` as it's synchronously returning a value, `async` would asynchronously execute the closure and return early.

Reply:
I think a better way to say it is that it blocks the current thread, we also first check that whether we're on the main thread because it will crash if we try to dispatch it in this way if we're already on that thread.

Reply:
Ah okay gotcha 👍